File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.py
class MyStack:

    def __init__(self):
        self.q1 = deque()
        self.q2 = deque()
        
    def push(self, x: int) -> None:
        self.q2.append(x)
        count = 0
        length = len(self.q1)
        while count < length:
            self.q2.append(self.q1.popleft())
            count += 1
        self.q1 = self.q2
        self.q2 = deque()

    def pop(self) -> int:
        return self.q1.popleft()

    def top(self) -> int:
        return self.q1[0]

    def empty(self) -> bool:
        return len(self.q1) == 0


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.java
class MyStack {
    Deque<Integer> q1;
    Deque<Integer> q2;
    
    public MyStack() {
        q1 = new ArrayDeque<>();
        q2 = new ArrayDeque<>();
    }
    
    public void push(int x) {
        q2.addLast(x);
        while(!q1.isEmpty()){
            q2.addLast(q1.pollFirst());
        }
        q1 = q2;
        q2 = new ArrayDeque<>();
    }
    
    public int pop() {
        if(!q1.isEmpty()) {
            return q1.pollFirst();
        } else {
            return -1;
        }
    }
    
    public int top() {
        return q1.peekFirst();
    }
    
    public boolean empty() {
        return q1.isEmpty();
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */